frontend/src/metabase/visualizations/components/ChartSettingsWidgetPopover.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import TippyPopover from "metabase/components/Popover/TippyPopover";
+import ChartSettingsWidget from "./ChartSettingsWidget";
+
+import { PopoverRoot, PopoverTitle } from "./ChartSettingsWidgetPopover.styled";
+
+interface Widget {
+  id: string;
+  props: Record<string, unknown>;
+}
+
+interface ChartSettingsWidgetPopoverProps {
+  anchor: HTMLElement;
+  handleEndShowWidget: () => void;
+  widgets: [Widget];
+  currentWidget: Widget;
+  extraWidgetProps: [];
+}
+
+export const ChartSettingsWidgetPopover = ({
+  anchor,
+  handleEndShowWidget,
+  widgets,
+  currentWidget,
+  extraWidgetProps,
+}: ChartSettingsWidgetPopoverProps) => {
+  const popoverWidgets =
+    currentWidget &&
+    [widgets.find(widget => widget.id === currentWidget.id)]
+      .map(widget => {
+        if (widget) {
+          return (
+            <ChartSettingsWidget
+              key={`${widget.id}`}
+              {...widget}
+              props={{
+                ...widget.props,
+                ...currentWidget.props,
+              }}
+              hidden={false}
+              {...extraWidgetProps}
+            />
+          );
+        }
+      })
+      .filter(w => !!w);
+
+  return (
+    <TippyPopover
+      reference={anchor}
+      content={
+        <PopoverRoot>
+          <PopoverTitle>Settings</PopoverTitle>
+          {popoverWidgets}
+        </PopoverRoot>
+      }
+      visible={!!anchor && popoverWidgets.length > 0}
```

Comment:
Following from the above comment, this also gives an impression that `popoverWidgets` can contain a lot of widgets while in reality, it's only a single widget. Not the end of the world though, but I like the code to be as expressive as possible.

Reply:
Agree with `popoverWidgets` being misleading.